scapy/utils.py

```diff
@@ -1636,12 +1639,23 @@ def _read_block_shb(self):
             raise EOFError
 
         blocklen = struct.unpack(self.endian + "I", _blocklen)[0]
-        if blocklen < 16:
-            warning("Invalid SHB block length!")
+        if blocklen < 28:
+            warning(f"Invalid SHB block length ({blocklen})!")
             raise EOFError
-        options = self.f.read(blocklen - 16)
-        self._read_block_tail(blocklen)
+
+        # Major version must be 1
+        _major = self.f.read(2)
+        major = struct.unpack(self.endian + "H", _major)[0]
+        if major != 1:
+            warning(f"SHB Major version {major} unsupported !")
+            raise EOFError
+
+        # Skip minor version & section length
+        self.f.seek(10, 1)
+
+        options = self.f.read(blocklen - 28)
         self._read_options(options)
+        self._read_block_tail(blocklen)
```

Comment:
@guedou Here I swapped the position of `self._read_block_tail(blocklen)` unintentionally. I think it's not making a big difference, but it was another unnecessary change.
By the way, should I proceed to make the changes here and your other suggestions? If yes, how should I do it? squash to a single commit (via git rebase? or git reset --soft?), or apply incremental ones?